### ISSUE TITLE
So, how do I use it ? Getting "Cannot find module 'node-rpi-rgb-led-matrix'"

### ISSUE THREAD

**Issue by hzeller** [zeitungen/node-rpi-rgb-led-matrix#11]
So, I am a total nodejs dummy, so you probably roll your eyes.

I tried the following, but it didn't work, presumably because `npm` did not install the module in a central place. Running `npm i` as root does also not yield the right result.

Is there anything else I need to do ? I presume I might need to add a path where the actual module is compiled or something, but if so, it might be worthwhile pointing that out in the `README.md` for idiots like me :)

```bash
node-rpi-rgb-led-matrix$ npm i
# ... compiles and stuff

node-rpi-rgb-led-matrix$ cat > testit.js
var LedMatrix = require("node-rpi-rgb-led-matrix");

var matrix = new LedMatrix(16);
matrix.fill(255, 50, 100);
matrix.setPixel(0, 0, 0, 50, 255);
^D

node-rpi-rgb-led-matrix$ sudo nodejs testit.js
module.js:340
    throw err;
          ^
Error: Cannot find module 'node-rpi-rgb-led-matrix'
    at Function.Module._resolveFilename (module.js:338:15)
    at Function.Module._load (module.js:280:25)
...
```

**Comment by zeitungen** [comment]
Yes, it is a nodejs beginner error. It is an error about the `require` function.

I'm not remember perfectly the rules about the `require` function but it is something like that:
- `require` check the module name passed as parameter is nodejs module. You can find them onto nodejs doc: https://nodejs.org/docs/latest-v5.x/api/ for the latest version (the 5.x).
For example: `var fs = require("fs");` load into `fs` the filesystem nodejs module 
- If the parameter is not a nodejs module, it checks if exists in `node_modules` directory. The `node_modules` directory contains all dependencies your project have.
- If it is not in `node_modules`, `require` try to load from the path. For example: `var module = require("./myawesomemodume.js")`

So, in summary:
- you should put `node-rpi-rgb-led-matrix` directory into a `node_modules` directory
- or instead of use `var LedMatrix = require("node-rpi-rgb-led-matrix");`, you should use `var LedMatrix = require("./index.js");`

Hope my explanations are clear. Good luck ! ;)

**Comment by hzeller** [comment]
Ah, thanks. Testing with `./index.js` worked!
